File: mezzanine-compiler/src/main/java/com/anthonycr/mezzanine/source/ElementSource.java
package com.anthonycr.mezzanine.source;

import org.jetbrains.annotations.NotNull;

import javax.lang.model.element.Element;

import io.reactivex.Flowable;

/**
 * An interface representing a source of elements.
 * <p>
 * Created by anthonycr on 5/22/17.
 */
public interface ElementSource {

    /**
     * A stream of elements.
     *
     * @return a valid observable.
     */
    @NotNull
    Flowable<Element> createElementStream();
}

File: mezzanine-compiler/src/main/java/com/anthonycr/mezzanine/source/ElementSource.kt
package com.anthonycr.mezzanine.source

import io.reactivex.Flowable
import javax.lang.model.element.Element

/**
 * An interface representing a source of elements.
 *
 *
 * Created by anthonycr on 5/22/17.
 */
interface ElementSource {

    /**
     * A stream of elements.

     * @return a valid observable.
     */
    fun createElementStream(): Flowable<Element>
}
